lib/gssapi/krb5/arcfour.c

```diff
@@ -740,15 +740,15 @@ OM_uint32 _gssapi_unwrap_arcfour(OM_uint32 *minor_status,
 	return GSS_S_BAD_MIC;
     }
 
+    if (conf_state)
+	*conf_state = conf_flag;
+
     HEIMDAL_MUTEX_lock(&context_handle->ctx_id_mutex);
     omret = _gssapi_msg_order_check(context_handle->order, seq_number);
     HEIMDAL_MUTEX_unlock(&context_handle->ctx_id_mutex);
     if (omret)
```

Comment:
I think minor_status needs to be set to 0 before this return.  Otherwise there might be random garbage used by the caller.   We might consider setting *minor_status = 0 upon entry to this function instead of setting it at each error return site.

Reply:
This is a pre-existing issue, that isn't affected by this change